tools/ncf_constraints.py

```diff
@@ -0,0 +1,90 @@
+import re
+import subprocess
+import json
+import os.path
+import shutil
+import sys
+import os
+import codecs
+
+
+### Constraint checking function 
+
+def from_list( parameter, accepted_result):
+  return parameter in accepted_result
+
+def max_length( parameter, max_size):
+  return len(parameter) <= max_size
+
+def min_length( parameter, min_size):
+  return len(parameter) >= min_size
+
+def match_regex(parameter, regex):
+  match = re.search(regex, parameter)
+  return match is not None
+
+def not_match_regex(parameter, regex):
+  return not match_regex(parameter, regex)
+
+def check_parameter(parameter, constraint):
+  """Checks that a parameter is ok with the constraint passed as parameter"""
+
+  result = True
+  errors = [];
+
+  # First check 'string' constraints
+  if not constraint.get("allow_empty_string", False):
+    if match_regex(parameter, "^$"):
+      errors.append("allow_empty_string")
+      result = False
+
+  if not constraint.get("allow_whitespace_string", False):
+    if match_regex(parameter, r'^\s') or match_regex(parameter, r'\s+$'):
```

Comment:
I don't like pure english sentence here since you need to fully read it to understand it.
I'd prefer something like "Default: leading/trailing whitespace forbiden"